lib/facter/warranty.rb

```diff
@@ -28,16 +28,20 @@ def create_dell_warranty_cache(cache)
     ['GetAssetWarrantyResult'] \
     ['Response']['DellAsset']['Warranties']['Warranty'].each do |w|
       end_date = Time.parse(w['EndDate'])
-      if end_date > Time.now()
+      if expiration_date < end_date
         warranty = true
-        expiration_date = end_date.strftime("%Y-%m-%d")
+        expiration_date = end_date
       end
   end
   rescue
   end
 
+  if expiration_date > Time.now()
```

Comment:
this can be reduced to one line `warranty = true if expiration_date > Time.now()`